include/particle.tcc

```diff
@@ -242,6 +242,38 @@ void mpm::Particle<Tdim>::initialise() {
   this->properties_["displacements"] = [&]() { return displacement(); };
 }
 
+//! Clone material state variables from neighbour particle
+template <unsigned Tdim>
+bool mpm::Particle<Tdim>::clone_neighbour_material_state_vars(
+    const mpm::dense_map& state_vars,
+    const std::shared_ptr<mpm::Material<Tdim>>& material) {
+  bool status = true;
```

Comment:
```suggestion
  bool status = false;
  if (material != nullptr) {
    if (this->material_id_ == material->id() ||
        this->material_id_ == std::numeric_limits<unsigned>::max()) {
      material_ = material;
      // Clone state variables
      auto mat_state_vars = material_->initialise_state_variables();
      if (mat_state_vars.size() == state_vars.size()) {
        this->state_variables_ = state_vars;
        status = true;
      } 
    }
  return status;
```

Reply:
I'm trying to avoid having throws, as C++ standard is changing to use `throw` only when you can recover from an error. Here, if we have any of the conditions failing we can't recover from it, so no reason to `throw`.

Reply:
okay